docs/source/migration/migration-to-0.8.rst

```diff
@@ -0,0 +1,47 @@
+Migration to 0.8.x
+==================
+
+Version 0.8.0 is is the first version which cleaned up the API. It introduced
+several breaking changes:
+
+- lifted transformers, deprecated in 0.7.0, got removed in favor of partial
+  transformers. Since 0.7.0 migration guide from lifted to partial is available
+  in ``partial transformers`` migrating from :ref:`lifted section<migrating-from-lifted>`.
+- ``.enableUnsafeOption`` option was removed - if ``Option`` unwrapping is
+  needed, it is recommended to use :ref:`partial transformers<partial-transformers>`
+- types returned by automatic derivation got split from types that are used
+  for user-provided transformations and configured (semiautomatic) derivation:
+  ``Transformer`` got split into ``Transformer`` and ``Transformer.AutoDerived``
+  while ``PartialTransformer`` got split into ``PartialTransformer`` and
+  ``PartialTransformer.Autoderived``.
+
+  It was caused be the change in mechanism for recursive derivation: since it
+  avoid boxing and allocation where possible, it used to check if summoned
+  implicit was generated by automatic derivation. If it came from
+  ``Transformer.derive`` or ``PartialTransformer.derive`` it was discarded and
+  macro attempted to derive it again without wrapping result in a type class.
+
+  It both complicated code as well as increased compilation times as each field
+  or subtype would attempt to summon implicit and then discard it if it didn't
+  came from user. Splitting types allows compiler to not summon any implicit if
+  user haven't provided any.
+
+  The consequence is only visible if there is some ``implicit def`` which takes
```

Comment:
This paragraph is quite hard to comprehend. Maybe some short example would do the job much better?

Reply:
Rewritten